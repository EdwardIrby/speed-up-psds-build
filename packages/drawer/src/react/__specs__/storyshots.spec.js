import path from 'path'
import initStoryshots, {
  snapshotWithOptions
} from '@storybook/addon-storyshots'

function createNodeMock(element) {
  if (element.type === 'div') {
    return { style: {} }
  }
  return null
}

initStoryshots({
  configPath: path.resolve(__dirname, '..', '..', '..', '.storybook'),
  test: snapshotWithOptions({
    createNodeMock
  })
})
