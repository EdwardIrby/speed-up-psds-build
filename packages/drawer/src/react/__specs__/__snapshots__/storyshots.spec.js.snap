// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots drawer controlled 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div>
      <button
        data-css-k922kt=""
        disabled={false}
        onClick={[Function]}
        style={Object {}}
      >
        <span
          data-css-15b13by=""
        >
          Toggle drawer
        </span>
      </button>
      <div
        className="css-1fq8nx6"
      >
        <div
          className="css-i9wfit"
          onClick={[Function]}
        >
          <div
            className="css-euzj8"
          >
            <div>
              Drawer Base Here
            </div>
          </div>
          <div
            className="css-qs2rg7"
          >
            <button
              aria-label="Collapse"
              className="css-xipskh"
              onClick={[Function]}
            >
              <div
                className="css-xnveog"
              >
                <div
                  data-css-otejxm=""
                >
                  <svg
                    aria-label="caret down icon"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
        <div
          className="css-ule0ur"
        >
          <div
            className="css-1j3g0qd"
          >
            Drawer Content Here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer default 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div
      className="css-1fq8nx6"
    >
      <div
        className="css-1oot5vc"
        onClick={[Function]}
      >
        <div
          className="css-euzj8"
        >
          <div>
            Drawer Base Here
          </div>
        </div>
        <div
          className="css-qs2rg7"
        >
          <button
            aria-label="Expand"
            className="css-xipskh"
            onClick={[Function]}
          >
            <div
              className="css-1byjzqg"
            >
              <div
                data-css-otejxm=""
              >
                <svg
                  aria-label="caret down icon"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        className="css-1i2htk9"
      >
        <div
          className="css-1j3g0qd"
        >
          Drawer Content Here
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer row component with actions 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div
      className="css-1fq8nx6"
    >
      <div
        className="css-1oot5vc"
        onClick={[Function]}
      >
        <div
          className="css-euzj8"
        >
          <div
            data-css-vl7fes=""
          >
            <div
              data-css-js3ti3=""
            >
              <span
                data-css-15yq56v=""
              >
                <a
                  href="https://duckduckgo.com?q=image"
                >
                  <img
                    src="https://cataas.com/cat"
                  />
                </a>
              </span>
              <div
                data-css-t9uarq=""
              >
                <span
                  data-css-1fxa3fl=""
                  onBlur={[Function]}
                  onFocus={[Function]}
                >
                  <a
                    href="https://duckduckgo.com?q=overlay"
                  >
                    Overlay
                  </a>
                </span>
              </div>
            </div>
            <div
              data-css-kxulf3=""
            >
              <div
                data-css-1jlgfae=""
              >
                <span
                  data-css-1le2bq0=""
                >
                  <a
                    href="https://duckduckgo.com?q=title"
                  >
                    I'm a Row with Actions
                  </a>
                </span>
              </div>
              <div
                data-css-cnfu1f=""
              >
                <span
                  data-css-1kcrbi9=""
                >
                  <span
                    data-css-1le2bq0=""
                  >
                    <a
                      href="https://duckduckgo.com?q=cats"
                    >
                      Kitten McCatbuns
                    </a>
                  </span>
                </span>
                <span
                  aria-hidden={true}
                  data-css-kykfz1=""
                >
                  Â·
                </span>
                <span
                  data-css-1kcrbi9=""
                >
                  23 hours of cuteness
                </span>
              </div>
            </div>
            <div
              data-css-1akhza9=""
            >
              <button
                data-css-2qvjr2=""
                onClick={[Function]}
              >
                <div
                  data-css-otejxm=""
                >
                  <svg
                    aria-label="more icon"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6 14.5c-1.1046 0-2-.8954-2-2s.8954-2 2-2 2 .8954 2 2-.8954 2-2 2zm12 0c-1.1046 0-2-.8954-2-2s.8954-2 2-2 2 .8954 2 2-.8954 2-2 2zm-6 0c-1.1046 0-2-.8954-2-2s.8954-2 2-2 2 .8954 2 2-.8954 2-2 2z"
                    />
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          className="css-qs2rg7"
        >
          <button
            aria-label="Expand"
            className="css-xipskh"
            onClick={[Function]}
          >
            <div
              className="css-1byjzqg"
            >
              <div
                data-css-otejxm=""
              >
                <svg
                  aria-label="caret down icon"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        className="css-1i2htk9"
      >
        <div
          className="css-1j3g0qd"
        >
          Drawer Content Here
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer stack of drawers 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div>
      <div
        className="css-1fq8nx6"
      >
        <div
          className="css-1oot5vc"
          onClick={[Function]}
        >
          <div
            className="css-euzj8"
          >
            <div>
              The Drawer #1
            </div>
          </div>
          <div
            className="css-qs2rg7"
          >
            <button
              aria-label="Expand"
              className="css-xipskh"
              onClick={[Function]}
            >
              <div
                className="css-1byjzqg"
              >
                <div
                  data-css-otejxm=""
                >
                  <svg
                    aria-label="caret down icon"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
        <div
          className="css-1i2htk9"
        >
          <div
            className="css-1j3g0qd"
          />
        </div>
      </div>
      <div
        className="css-1fq8nx6"
      >
        <div
          className="css-1oot5vc"
          onClick={[Function]}
        >
          <div
            className="css-euzj8"
          >
            <div>
              The Drawer #2
            </div>
          </div>
          <div
            className="css-qs2rg7"
          >
            <button
              aria-label="Expand"
              className="css-xipskh"
              onClick={[Function]}
            >
              <div
                className="css-1byjzqg"
              >
                <div
                  data-css-otejxm=""
                >
                  <svg
                    aria-label="caret down icon"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
        <div
          className="css-1i2htk9"
        >
          <div
            className="css-1j3g0qd"
          />
        </div>
      </div>
      <div
        className="css-1fq8nx6"
      >
        <div
          className="css-1oot5vc"
          onClick={[Function]}
        >
          <div
            className="css-euzj8"
          >
            <div>
              The Drawer #2
            </div>
          </div>
          <div
            className="css-qs2rg7"
          >
            <button
              aria-label="Expand"
              className="css-xipskh"
              onClick={[Function]}
            >
              <div
                className="css-1byjzqg"
              >
                <div
                  data-css-otejxm=""
                >
                  <svg
                    aria-label="caret down icon"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
        <div
          className="css-1i2htk9"
        >
          <div
            className="css-1j3g0qd"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer stack of non-sibling drawers 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div>
      <div>
        <div
          className="css-1fq8nx6"
        >
          <div
            className="css-1oot5vc"
            onClick={[Function]}
          >
            <div
              className="css-euzj8"
            >
              <div>
                The Drawer #1
              </div>
            </div>
            <div
              className="css-qs2rg7"
            >
              <button
                aria-label="Expand"
                className="css-xipskh"
                onClick={[Function]}
              >
                <div
                  className="css-1byjzqg"
                >
                  <div
                    data-css-otejxm=""
                  >
                    <svg
                      aria-label="caret down icon"
                      role="img"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                      />
                    </svg>
                  </div>
                </div>
              </button>
            </div>
          </div>
          <div
            className="css-1i2htk9"
          >
            <div
              className="css-1j3g0qd"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          className="css-1fq8nx6"
        >
          <div
            className="css-1oot5vc"
            onClick={[Function]}
          >
            <div
              className="css-euzj8"
            >
              <div>
                The Drawer #2
              </div>
            </div>
            <div
              className="css-qs2rg7"
            >
              <button
                aria-label="Expand"
                className="css-xipskh"
                onClick={[Function]}
              >
                <div
                  className="css-1byjzqg"
                >
                  <div
                    data-css-otejxm=""
                  >
                    <svg
                      aria-label="caret down icon"
                      role="img"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                      />
                    </svg>
                  </div>
                </div>
              </button>
            </div>
          </div>
          <div
            className="css-1i2htk9"
          >
            <div
              className="css-1j3g0qd"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          className="css-1fq8nx6"
        >
          <div
            className="css-1oot5vc"
            onClick={[Function]}
          >
            <div
              className="css-euzj8"
            >
              <div>
                The Drawer #2
              </div>
            </div>
            <div
              className="css-qs2rg7"
            >
              <button
                aria-label="Expand"
                className="css-xipskh"
                onClick={[Function]}
              >
                <div
                  className="css-1byjzqg"
                >
                  <div
                    data-css-otejxm=""
                  >
                    <svg
                      aria-label="caret down icon"
                      role="img"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                      />
                    </svg>
                  </div>
                </div>
              </button>
            </div>
          </div>
          <div
            className="css-1i2htk9"
          >
            <div
              className="css-1j3g0qd"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer using custom aria label 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div
      className="css-1fq8nx6"
    >
      <div
        className="css-1oot5vc"
        onClick={[Function]}
      >
        <div
          className="css-euzj8"
        >
          <div>
            Drawer Base Here
          </div>
        </div>
        <div
          className="css-qs2rg7"
        >
          <button
            aria-label="Expand custom drawer"
            className="css-xipskh"
            onClick={[Function]}
          >
            <div
              className="css-1byjzqg"
            >
              <div
                data-css-otejxm=""
              >
                <svg
                  aria-label="caret down icon"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        className="css-1i2htk9"
      >
        <div
          className="css-1j3g0qd"
        >
          Drawer Content Here
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer using onToggle prop 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div>
      <div
        className="css-1fq8nx6"
      >
        <div
          className="css-1oot5vc"
          onClick={[Function]}
        >
          <div
            className="css-euzj8"
          >
            <div>
              Drawer is 
              closed
            </div>
          </div>
          <div
            className="css-qs2rg7"
          >
            <button
              aria-label="Expand"
              className="css-xipskh"
              onClick={[Function]}
            >
              <div
                className="css-1byjzqg"
              >
                <div
                  data-css-otejxm=""
                >
                  <svg
                    aria-label="caret down icon"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
        <div
          className="css-1i2htk9"
        >
          <div
            className="css-1j3g0qd"
          >
            Drawer Content Here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer using startOpen prop 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div
      className="css-1fq8nx6"
    >
      <div
        className="css-i9wfit"
        onClick={[Function]}
      >
        <div
          className="css-euzj8"
        >
          <div>
            Base here
          </div>
        </div>
        <div
          className="css-qs2rg7"
        >
          <button
            aria-label="Collapse"
            className="css-xipskh"
            onClick={[Function]}
          >
            <div
              className="css-xnveog"
            >
              <div
                data-css-otejxm=""
              >
                <svg
                  aria-label="caret down icon"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        className="css-ule0ur"
      >
        <div
          className="css-1j3g0qd"
        >
          <div>
            Panel here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots drawer with row component 1`] = `
<div
  data-css-claeju=""
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "padding": 48,
      }
    }
  >
    <div
      className="css-1fq8nx6"
    >
      <div
        className="css-1oot5vc"
        onClick={[Function]}
      >
        <div
          className="css-euzj8"
        >
          <div
            data-css-vl7fes=""
          >
            <div
              data-css-js3ti3=""
            >
              <div
                data-css-y8jh93=""
              />
            </div>
            <div
              data-css-kxulf3=""
            >
              <div
                data-css-1jlgfae=""
              >
                Look at me! I'm a &lt;Row /&gt;!
              </div>
              <div
                data-css-cnfu1f=""
              >
                <span
                  data-css-1kcrbi9=""
                >
                  Kitten McCatbuns
                </span>
                <span
                  aria-hidden={true}
                  data-css-kykfz1=""
                >
                  Â·
                </span>
                <span
                  data-css-1kcrbi9=""
                >
                  23 hours of cuteness
                </span>
              </div>
            </div>
            <div
              data-css-okadky=""
            >
              <button
                data-css-2qvjr2=""
              >
                <div
                  data-css-otejxm=""
                >
                  <svg
                    aria-label="more icon"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6 14.5c-1.1046 0-2-.8954-2-2s.8954-2 2-2 2 .8954 2 2-.8954 2-2 2zm12 0c-1.1046 0-2-.8954-2-2s.8954-2 2-2 2 .8954 2 2-.8954 2-2 2zm-6 0c-1.1046 0-2-.8954-2-2s.8954-2 2-2 2 .8954 2 2-.8954 2-2 2z"
                    />
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          className="css-qs2rg7"
        >
          <button
            aria-label="Expand"
            className="css-xipskh"
            onClick={[Function]}
          >
            <div
              className="css-1byjzqg"
            >
              <div
                data-css-otejxm=""
              >
                <svg
                  aria-label="caret down icon"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 15.41l-5-5L8.41 9 12 12.58 15.59 9 17 10.41"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        className="css-1i2htk9"
      >
        <div
          className="css-1j3g0qd"
        >
          Drawer Content Here
        </div>
      </div>
    </div>
  </div>
</div>
`;
